src/Extensions/CacheKeyExtension.php

```diff
@@ -13,9 +14,14 @@
 
 /**
  * @property DataObject|$this $owner
+ * @method HasManyList|CacheKey CacheKeys()
  */
 class CacheKeyExtension extends DataExtension
 {
+    private static array $has_many = [
```

Comment:
Would it not be a 1:1 relation?

Reply:
Oh, I'm guessing if you try to create an additional one you get an exception which we'd like to prevent so we're doing a `has_many` here? If so we should add this as a comment so anyone else reading the code has context

Reply:
Yea, Silverstripe only supports a `has_many` polymorphic relationship, as it doesn't itself define a unique index on the "other side" of the relationship. So we could technically create many `CacheKey` records with the same `RecordClass` and `RecordID`.

I will add a note to explain this though 😄 We have one in the `remove()` method, but it's not very likely for a dev to read that.